test/integration/Integration.Contracts/ItemCommandsTopic.cs

```diff
@@ -1,20 +1,20 @@
 ﻿namespace Integration.Contracts
 {
     using PetProjects.Framework.Kafka.Contracts.Topics;
+    using TopicProperties;
 
-    public class ItemCommandsTopic : ITopic<ItemCommandsV1>
+    public class ItemCommandsTopic : TopicBase<ItemCommandsV1>
     {
-        public ItemCommandsTopic()
+        public ItemCommandsTopic(string environment)
+            : base(environment)
         {
         }
 
-        public string TopicFullName => this.SetTopicName().TopicFullName;
-
-        public TopicBuilder SetTopicName()
+        public override TopicBuilder SetTopicName()
         {
-            return new TopicBuilder($"{typeof(ItemCommandsV1).FullName}", MessageType.Commands)
-                .WithApplication("console-app-test")
-                .WithVersion(1);
+            return new TopicBuilder(new Item(), MessageType.Commands, this.Environment)
+                        .WithApplication(new TestApp())
```

Comment:
This needs to be passed by parameter somewhere. I suggest passing it via ctor like environment.

Reply:
Also I suggest creating a GenericApplication with a ctor that takes a string so that a user can implement a consumer like this:

```
public class ConcreteConsumer : Consumer
{
    public ConcreteConsumer (config) : base(new ItemCommandsTopic(config.Environment, new GenericApplication(config.Application)) {}
}
```

Actually is it really useful to have an IApplication interface? Why not just use a string?

Reply:
Think back, we don't need the fluent stuff. We could do this by having only ctors. I will change this in order to simplify it.